Added timespec_div

GPaso opened the pull request:
I found this library quite useful and I used some functions, but I also needed an extra one that i couldn't find so here it is.

I needed to divide a timespec by an integer value in order to have a mean value from a sum of timespecs.

I hope it's helpful and Thanks!

G.Pasotti

GPaso committed:
Added timespec_div

Needed to divide a timespec by an integer number

GPaso committed:
Added timespec_div

GPaso committed:
Update README.md

added timespec_div description

GPaso committed:
Update timespec.c

fixed some errors in TEST_DIV_FUNC

GPaso committed:
Update timespec.c

Fixes for 32 bits systems

GPaso committed:
Update timespec.c

Fixes for 32 bits

GPaso committed:
Update timespec.c

Fixes

GPaso committed:
Update timespec.c

Rewrite the function to make it work for 32 bits

GPaso commented:
Sorry for the mess, i have some issues making it work for i386, do you have any suggestion to make it work without losing precision?

GPaso committed:
Update timespec.c

solemnwarning commented:
Hi G.Pasotti,

Thanks for this, mostly looks good to me although I'd rename it to something like `timespec_idiv()` since there is already a `timespec_mod()` function which takes two timespec structures as operands rather than a timespec and an integer.

I haven't tested it, I think something like this might handle the rounding correctly:

```
unsigned long extra_nsec = rem_sec * NSEC_PER_SEC;
result.tv_nsec = (ts.tv_nsec + extra_nsec) / divisor;
```

I could also be completely wrong.

Daniel

GPaso committed:
Update timespec.c

changed name from timespec_div to timespec_idiv and test fix for fixing the rounding error

GPaso committed:
Update README.md

changed timespec_div to timespec_idiv

GPaso committed:
Update timespec.h

changed name from timespec_div to timespec_idiv

GPaso commented:
Your suggestion worked! Thanks!
Do you think i should add more test cases? or anything else?
